### PR TITLE
Feature/developer warning

### PR DESCRIPTION
**Pull request by chandlerprall**

### CODE REVIEW

**Comment by chandlerprall** [comment]
@hartzis @michaelmerrill any thoughts here?

File: index.js [hunk 1]
```diff
@@ -1,10 +1,56 @@
+var HAS_PROCESS_ENV_NODE_ENV = typeof process !== 'undefined' && Object.prototype.hasOwnProperty(process, 'env') && Object.prototype.hasOwnProperty(process.env, 'NODE_ENV');
+var IS_PRODUCTION = HAS_PROCESS_ENV_NODE_ENV && process.env.NODE_ENV === 'production';
+var Immutable = require('immutable');
```
Comment (hartzis) [inline index.js]: move `immutable` related functions to new file?

File: index.js [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+
+function isImmutable(value) {
+    // immutable v4 added `isImmutable`
+    if (Immutable.hasOwnProperty(isImmutable)) {
+        return Immutable.isImmutable(value);
+    } else {
+        return Immutable.Map.isMap(value) ||
```
Comment (hartzis) [inline index.js:10]: these are the only supported immutable types? record?

File: index.js [hunk 3]
```diff
@@ -1,10 +1,56 @@
+var HAS_PROCESS_ENV_NODE_ENV = typeof process !== 'undefined' && Object.prototype.hasOwnProperty(process, 'env') && Object.prototype.hasOwnProperty(process.env, 'NODE_ENV');
+var IS_PRODUCTION = HAS_PROCESS_ENV_NODE_ENV && process.env.NODE_ENV === 'production';
+var Immutable = require('immutable');
+
+function isImmutable(value) {
+    // immutable v4 added `isImmutable`
+    if (Immutable.hasOwnProperty(isImmutable)) {
+        return Immutable.isImmutable(value);
```
Comment (hartzis) [inline index.js:8]: cool

**Comment by hartzis** [comment]
test(s)? maybe?

**Review by hartzis — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (chandlerprall) [reply index.js:10]: The others don't have `isX` methods

**Pull request merged**